Error building from master caused by namespace issue

**Godot version:**
Master

**OS/device including version:**
Ubuntu 16.04
gcc version 5.5.0


**Issue description:**
Building from master with ``scons p=x11 target=release_debug -j4"

Error produced is:
```
editor/editor_layouts_dialog.cpp: In constructor 'EditorLayoutsDialog::EditorLayoutsDialog()':
editor/editor_layouts_dialog.cpp:123:42: error: 'ItemList::SelectMode' is not a class or namespace
  layout_names->set_select_mode(ItemList::SelectMode::SELECT_MULTI);
```

Introduced by https://github.com/godotengine/godot/commit/9986cf70058011fed81c8a2bba4df47e3e58b3ba

Appears to be same use of namespaces as https://github.com/godotengine/godot/issues/28634

I looked online and it appears it is a weird thing that is best supported on windows and only on certain versions of GCC

--- a/editor/editor_layouts_dialog.cpp
+++ b/editor/editor_layouts_dialog.cpp
@@ -120,7 +120,7 @@ EditorLayoutsDialog::EditorLayoutsDialog() {
 	layout_names->set_anchor_and_margin(MARGIN_LEFT, ANCHOR_BEGIN, 5);
 	layout_names->set_anchor_and_margin(MARGIN_RIGHT, ANCHOR_END, -5);
 	layout_names->set_v_size_flags(Control::SIZE_EXPAND_FILL);
-	layout_names->set_select_mode(ItemList::SelectMode::SELECT_MULTI);
+	layout_names->set_select_mode(ItemList::SELECT_MULTI);
 	layout_names->set_allow_rmb_select(true);
 
 	name = memnew(LineEdit);

